lib/kernel.g

```diff
@@ -168,7 +168,7 @@ POSITION_NOT := function( arg )
                 return i;
             fi;
         od;
-        return LENGTH(arg[1]) + 1;
+        return arg[3] + 1;
```

Comment:
This is not quite correct: Think about arg[3] being 0 (in fact, `PositionNot(list,x)` is defined as `PositionNot(list,x,0)`.

So instead of changing this line, you could insert this before line 166:
```gap
if arg[3]+1 > LENGTH(arg[1]) then
    return arg[3] + 1;
fi;
```